### ISSUE TITLE
V4.2.0 introduced issue in cypher generation

### ISSUE THREAD

**Issue by mathix420** [neo4j/graphql#4077]
**Describe the bug**
Don't exactly know what's happening, but it seems to be related with authorization.

**Type definitions**
```graphql
type PreviewClip @mutation(operations: [DELETE]) {
    id: ID! @id
    startTime: Int!
    duration: Int!

    markedAsDone: Boolean! @default(value: false)

    clippedFrom: Video! @relationship(type: "VIDEO_HAS_PREVIEW_CLIP", direction: IN)

    creationDate: DateTime! @timestamp(operations: [CREATE])
    lastUpdate: DateTime! @timestamp(operations: [CREATE, UPDATE])
}
# Rules
extend type PreviewClip
    @authorization(
        filter: [
            { where: { node: { clippedFrom: { publisher: { id: "$jwt.sub" } } } } }
            { where: { jwt: { roles_INCLUDES: "admin" } } }
        ]
    )

type Video @mutation(operations: [UPDATE]) {
    id: ID! @id

    publisher: User! @relationship(type: "PUBLISHER", direction: IN)

    creationDate: DateTime! @timestamp(operations: [CREATE])
    lastUpdate: DateTime! @timestamp(operations: [CREATE, UPDATE])
}
# Rules
extend type Video
    @authorization(
        filter: [
            { where: { node: { publisher: { id: "$jwt.sub" } } } }
            { where: { jwt: { roles_INCLUDES: "admin" } } }
            {
                requireAuthentication: false
                operations: [READ]
                where: { node: { processing: "published" } }
            }
        ]
    )
    
type User @mutation(operations: [UPDATE]) {
    id: ID! @id
}

# Rules
extend type User
    @authorization(
        validate: [
            { operations: [UPDATE], where: { node: { id: "$jwt.sub" } } }
            { operations: [UPDATE], where: { jwt: { roles_INCLUDES: "admin" } } }
        ]
    )
```

**To Reproduce**

1. Execute the following Query
    ```graphql
    query listPossiblePreviewClips($videoId: ID!) {
      previewClips(where: { clippedFrom: { id: $videoId }, NOT: { markedAsDone: true } }) {
        id
      }
    }
    ```
2. See error
```
{
  message: 'Variable `clippedFromCount` not defined (line 11, column 9 (offset: 405))\n' +
    '"WHERE ((clippedFromCount <> 0 AND this0.id = $param1) AND (($isAuthenticated = true AND var3 = true) OR ($isAuthenticated = true AND $param3 IN $jwt.roles)))"\n' +
    '         ^',
  locations: [ { line: 2, column: 3 } ],
  path: [ 'previewClips' ],
  extensions: {
    code: 'INTERNAL_SERVER_ERROR',
    stacktrace: [
      'Neo4jError: Variable `clippedFromCount` not defined (line 11, column 9 (offset: 405))',
      '"WHERE ((clippedFromCount <> 0 AND this0.id = $param1) AND (($isAuthenticated = true AND var3 = true) OR ($isAuthenticated = true AND $param3 IN $jwt.roles)))"',
      '         ^',
      '',
      '    at captureStacktrace (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/result.js:613:17)',
      '    at new Result (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/result.js:105:23)',
      '    at newCompletedResult (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:510:12)',
      '    at Object.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:342:20)',
      '    at Transaction.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:183:34)',
      '    at ManagedTransaction.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction-managed.js:56:21)',
      '    at Executor.transactionRun (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/dist/classes/Executor.js:162:28)',
      '    at /home/mathix/projects/peach_api/node_modules/@neo4j/graphql/dist/classes/Executor.js:143:33',
      '    at TransactionExecutor._safeExecuteTransactionWork (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:213:26)',
      '    at TransactionExecutor.<anonymous> (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:200:46)'
    ]
  }
}
```

**Expected behavior**
Work as expected in 4.1.3

**System (please complete the following information):**

* OS: Archlinux
* Version: @neo4j/graphql@4.2.0
* Node.js version: v18.16.1

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @mathix420. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @mathix420. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @mathix420! :pray: We will now prioritise the bug and address it appropriately.

**Comment by angrykoala** [comment]
Hi @mathix420 
This one is caused by the new changes to the cypher translation. I'll take a look to get this fixed soon

**Comment by mathix420** [comment]
Thanks for the feedback @angrykoala !

**Comment by angrykoala** [comment]
@mathix420 
This fix will be available in the next release `4.3.1`

**Comment by mathix420** [comment]
Amazing, thanks for the quick fix!
